src/main/java/core/stack/MateStack.java

```diff
@@ -1,19 +1,50 @@
 package core.stack;
 
+import java.util.EmptyStackException;
+
 public class MateStack<T> {
-    public void push(T value) {
+    private static final float LOAD_COEFFICIENT = 1.5f;
+    private static final int DEFAULT_CAPACITY = 10;
+    private T[] stack;
+    private int size;
 
+    public MateStack() {
+        size = 0;
+        stack = (T[]) new Object[DEFAULT_CAPACITY];
+    }
+
+    public void push(T value) {
+        if (size >= stack.length) {
+            grow();
+        }
+        stack[size] = value;
+        size++;
     }
 
     public T peek() {
-        return null;
+        return getLastItem();
     }
 
     public T pop() {
-        return null;
+        T lastItem = getLastItem();
+        stack[size - 1] = null;
+        size--;
+        return lastItem;
     }
 
     public int size() {
-        return 0;
+        return size;
+    }
+
+    private void grow() {
+        int newLength = (int) (LOAD_COEFFICIENT * stack.length);
+        System.arraycopy(stack, 0, stack, 0, stack.length);
```

Comment:
please create the new array with the increased size
then copy the number of elements that is present in the old array, to the new array